kernel.py

```diff
@@ -79,9 +80,15 @@ def __init__(self, connection_filename="kernel.json", connection_fn_with_pid=Tru
         self._lock = threading.Lock()
         self._condition = threading.Condition(lock=self._lock)
 
+        # Get connection file path
         if connection_fn_with_pid:
             name, ext = os.path.splitext(connection_filename)
             connection_filename = "%s-%i%s" % (name, os.getpid(), ext)
+            try:
+                from jupyter_core.paths import jupyter_runtime_dir
+                connection_filename = os.path.join(jupyter_runtime_dir(), connection_filename)
```

Comment:
I think we should not add another directory on `connection_filename` if the user provided some own custom `connection_filename`.
Rather, we should change the default `connection_filename` to `None`, and then in that case, if also `connection_fn_with_pid`, then we can have this logic, which falls back to the old default `connection_filename = "kernel.json"`.